src/test/java/org/openmrs/contrib/qaframework/page/PatientVisitsDashboardPage.java

```diff
@@ -45,8 +50,28 @@ public String getPageUrl() {
 		return "coreapps/patientdashboard/patientDashboard.page";
 	}
 
-	public void goToCaptureVitals() {
-		findElement(CAPTURE_VITALS).click();
+	public PatientCaptureVitalsPage goToPatientCaptureVitalsPage() {
+		waiter.until(ExpectedConditions.presenceOfElementLocated(CAPTURE_VITALS));
+		clickOn(CAPTURE_VITALS);
+		return new PatientCaptureVitalsPage(this);
+	}
+	
+	public EditVitalsPage goToEditVitalsPage() {
+		waiter.until(ExpectedConditions.presenceOfElementLocated(EDIT_VITALS_ICON));
```

Comment:
@kdaud just thought in cases of slow connection, they would load slowly hence the waiter

Reply:
> just thought in cases of slow connection, they would load slowly hence the waiter

invoking waiter call suits when the element takes long to be rendered on the page otherwise we can't cater for slow connection using waiter method.